content/en/docs/concepts/services-networking/service.md

```diff
@@ -945,20 +945,20 @@ API object can be found at:
 
 ### TCP
 
-{{< feature-state for_k8s_version="v1.0" state="stable" >}}
+{{< feature-state feature_name="tcp" for_k8s_version="v1.0" state="stable" >}}
```

Comment:
I was kind of expecting this to be
```markdown
{{< feature-state feature_name="service:tcp" for_k8s_version="v1.0" state="stable" >}}
```

Thinking of the SCTP example later, Kubernetes has alpha support for SCTP Services. That's not the same as alpha support for SCTP overall.

If, say, we added SCTP to Ingress, or we supported QUIC between Kubelet and the apiserver, I think we'd want to distinguish those protocols by where you can use them.

Does this make sense?

Reply:
I have not fully thought out how the feature names should be defined. I put together two examples as a test. The feature names in these examples should be changed to suit. My first thought is to let the authors determine/validate the feature name, if it is not obvious. I guess at some point the names possibly could be localized.

Reply:
The [feature gate](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/) names are all in English at the moment, and as they are things that people interact with (eg via the `kubectl` command line) I would explain them but not localize them.

Is there a way to link the Hugo `feature-state` shortcodes with the feature gates that get compiled into the the built binaries? If that's a step too far (and I suspect it is), what's the easy change for now that lets us get there eventually?

Reply:
Right about the localization. Ideally there would be coordination between the feature-state labels and the actual feature gates. Not sure how to get there. Also, there is the feature gates page in the reference section. Typically, the labels are manually updated by the author of the code feature when the feature is upgraded. One idea about naming convention: create a root_feature_name in the page front matter and use that to scope the actual feature_name, such as service:tcp where service is the root_feature_name. For now, it would be straightforward to update pages that use the feature-state shortcode with a feature_name.